docs/advanced/ecdsa.md

```diff
@@ -0,0 +1,46 @@
+KICL supports the ECDSA-NIST256P-CHALLENGE SASL authentication mechanism. This document describes both how the mechanism works and how the library can be used to authenticate with this method.
+
+### Protocol description
+
+The ECDSA authentication mechanism makes use of the SASL extension. This must be requested by the client prior to any authentication attempts (`CAP REQ :sasl`). Once the server sends an `ACK`, authentication can begin.
+
+The mechanism makes use of the NIST P-256 curve (`NID_X9_62_prime256v1` in OpenSSL) which (per the name) is defined over a 256 bit prime finite field. The parameters for this curve are available [from the NSA](https://www.nsa.gov/ia/_files/nist-routines.pdf) in section 4.3. The curve itself doesn't fit the [SafeCurves requirements](http://safecurves.cr.yp.to/).
+
+Atheme is the only IRC services software that appears to implement this authentication method. Atheme's implementation requires that:
```

Comment:
I'm aware of their existence but don't really use them or know how they implement SASL. the only networks I'm on don't support SASL or use Atheme or Anope.